CondCore/SiStripPlugins/plugins/SiStripApvGain_PayloadInspector.cc

```diff
@@ -0,0 +1,420 @@
+#include "CondCore/Utilities/interface/PayloadInspectorModule.h"
+#include "CondCore/Utilities/interface/PayloadInspector.h"
+#include "CondCore/CondDB/interface/Time.h"
+
+// the data format of the condition to be inspected
+#include "CondFormats/SiStripObjects/interface/SiStripApvGain.h"
+
+#include <memory>
+#include <sstream>
+
+namespace {
+
+  /************************************************
+    1d histogram of SiStripApvGains of 1 IOV 
+  *************************************************/
+
+  // inherit from one of the predefined plot class: Histogram1D
+  class SiStripApvGainsValue : public cond::payloadInspector::Histogram1D<SiStripApvGain> {
+    
+  public:
+    SiStripApvGainsValue() : cond::payloadInspector::Histogram1D<SiStripApvGain>("SiStripApv Gains values",
+										 "SiStripApv Gains values", 200,0.2,1.8){
+      Base::setSingleIov( true );
+    }
+    
+    bool fill( const std::vector<std::tuple<cond::Time_t,cond::Hash> >& iovs ){
+      for ( auto const & iov: iovs) {
+	std::shared_ptr<SiStripApvGain> payload = Base::fetchPayload( std::get<1>(iov) );
+	if( payload.get() ){
+	 
+	  std::vector<uint32_t> detid;
+	  payload->getDetIds(detid);
+	  
+	  for (size_t id=0;id<detid.size();id++){
```

Comment:
hi @mmusich - please use
for ( const auto & d : detid) { 
   SiStripAprGain::range range=payload->getRange(d);
...